### PR TITLE
Fix: Undefined index for category column

### PR DESCRIPTION
**Pull request by mukeshpanchal27**
**Steps to replicate issue**

Admin Panel > Block Lab > Add New
Add Title then leave page
Post will auto save
Go to Admin Panel > Block Lab there it shows "Notice: Undefined index: in /var/www/html/wordpress/wp-content/plugins/block-lab/php/post-types/class-block-post.php on line 1160"

**Issue**

When we add new field without saving it means when field save by auto save then plugin generate undefined notice

**Screenshot**
Error

### CODE REVIEW

File: php/post-types/class-block-post.php
Comment (lukecarbis) [inline php/post-types/class-block-post.php]: Should we also include a check to see if the block has a category?
`if ( ! empty( $block->category ) )`

**Review by lukecarbis — CHANGES_REQUESTED** [review]
Great! Thanks for finding this. One small question before merging.

File: php/post-types/class-block-post.php
Comment (mukeshpanchal27) [reply php/post-types/class-block-post.php]: Yes it's working fine if we add above conditional logic. But isset is already used in some conditional logic so i have added isset instead of not empty condition.

`if ( isset( $this->controls[ $control_name ] ) ) {`
`if ( isset( $icons[ $block->icon ] ) ) {`

Comment (lukecarbis) [reply php/post-types/class-block-post.php]: The problem is occurring because `$block->category` is equal to an empty string, and `$categories['']` is not set.

Checking if `$categories['']` is set does help, so we should absolutely keep that, but I'm just suggesting that it would also nice to check right after we instantiate the `Block` whether the category is empty. If it is, we can skip running `get_block_categories`, `wp_list_pluck`, etc.

Comment (lukecarbis) [reply php/post-types/class-block-post.php]: Like this: 

```php
$block = new Block( $post_id );

if ( ! empty( $block->category ) ) {
    $categories = get_block_categories( get_post() );
    $categories = wp_list_pluck( $categories, 'title', 'slug' );

    if ( isset( $categories[ $block->category ] ) ) {
        echo esc_html( $categories[ $block->category ] );
    }
}
```

**Comment by mukeshpanchal27** [comment]
@lukecarbis Apply changes please check it and let us know in case of any issue

**Comment by lukecarbis** [comment]
@mukeshpanchal27 Thanks again! It looks like Travis has picked up some minor code styling problems with the PR – we can't merge it unless it passes all these tests.

- Looks like you've got unnecessary indentation on line 1118.
- Since we've moved the definition of `$categories` inside the conditional, line 1117 only needs a single space before the equals sign

**Review by lukecarbis — APPROVED** [review]
Tested, and could replicate the issue. This PR resolved it.

Thank you Mukesh!

**Pull request merged**
